src/aws_encryption_sdk/keyring/raw_keyring.py

```diff
@@ -0,0 +1,390 @@
+        )
+
+        # EncryptedData to EncryptedDataKey
+        encrypted_data_key = serialize_wrapped_key(
+            key_provider=self._key_provider,
+            wrapping_algorithm=self._wrapping_algorithm,
+            wrapping_key_id=self.key_name,
+            encrypted_wrapped_key=encrypted_wrapped_key,
+        )
+
+        # Update Keyring Trace
+        if encrypted_data_key:
+            keyring_trace = KeyringTrace(
+                wrapping_key=encrypted_data_key.key_provider, flags={KeyringTraceFlag.WRAPPING_KEY_ENCRYPTED_DATA_KEY}
+            )
+
+            # Add encrypted data key to encryption_materials
+            encryption_materials.add_encrypted_data_key(
+                encrypted_data_key=encrypted_data_key, keyring_trace=keyring_trace
+            )
+        return encryption_materials
+
+    def on_decrypt(self, decryption_materials, encrypted_data_keys):
+        # type: (DecryptionMaterials, Iterable[EncryptedDataKey]) -> DecryptionMaterials
+        """Attempt to decrypt the encrypted data keys.
+
+        :param decryption_materials: Decryption materials for the keyring to modify
+        :type decryption_materials: aws_encryption_sdk.materials_managers.DecryptionMaterials
+        :param encrypted_data_keys: List of encrypted data keys
+        :type: List of `aws_encryption_sdk.structures.EncryptedDataKey`
+        :returns: Optionally modified decryption materials
+        :rtype: aws_encryption_sdk.materials_managers.DecryptionMaterials
+        """
+        if decryption_materials.data_encryption_key:
+            return decryption_materials
+
+        # Decrypt data key
+        expected_key_info_len = len(self._key_info_prefix) + self._wrapping_algorithm.algorithm.iv_len
+        for key in encrypted_data_keys:
+            if (
+                key.key_provider.provider_id == self._key_provider.provider_id
+                and len(key.key_provider.key_info) == expected_key_info_len
+                and key.key_provider.key_info.startswith(self._key_info_prefix)
+            ):
+                # Wrapped EncryptedDataKey to deserialized EncryptedData
+                encrypted_wrapped_key = deserialize_wrapped_key(
+                    wrapping_algorithm=self._wrapping_algorithm,
+                    wrapping_key_id=self.key_name,
+                    wrapped_encrypted_key=key,
+                )
+                # EncryptedData to raw key string
+                try:
+                    plaintext_data_key = self._wrapping_key_structure.decrypt(
+                        encrypted_wrapped_data_key=encrypted_wrapped_key,
+                        encryption_context=decryption_materials.encryption_context,
+                    )
+
+                except Exception as error:  # pylint: disable=broad-except
+                    logger = logging.getLogger()
+                    logger.error(error.__class__.__name__, ":", str(error))
```

Comment:
Since this is a case where if we reach this point, we would reasonably expect to be able to decrypt this data key, I think the right approach here is to log this as an exception[1] with a message that identifies this keyring (ex[2]).

[1] https://docs.python.org/3/library/logging.html#logging.Logger.exception
[2] https://github.com/aws/aws-encryption-sdk-python/blob/35e6549dbd44c72f2c5ff44a53bf51ce9da63195/src/aws_encryption_sdk/key_providers/kms.py#L264-L267